src/auth/PrivateRoute.jsx

```diff
@@ -0,0 +1,20 @@
+import React, { useContext } from "react";
+import { Route, Redirect } from "react-router-dom";
+import { AuthContext } from "./Authentication";
+
+/**
+ * Use this higher-order component with the component that
```

Comment:
this is not a higher order component

Reply:
Ah I see. I was about to apply higher order component and wrap all the component that required the used to be authenticated, but then used this way. I was confused about these two way.